### PR TITLE
Added Optional Scope parameter to ClientCredentialProvider

### PR DESCRIPTION
**Pull request by FiromsaA**
Added new parameter to Client Credential Provider to allow custom scopes to be passed in. Issue #44.

### CODE REVIEW

**Review by MIchaelMainer — DISMISSED** [review]
Looks good. This brings up a couple of questions. Does setting the . default scope make sense for other providers? Also, are there any scenarios that always use OpenID Connect? If so, should we support any of the OpenID scopes for defaults?

File: src/Microsoft.Graph.Auth/ConfidentialClient/ClientCredentialProvider.cs [hunk 1]
```diff
@@ -16,6 +16,10 @@ namespace Microsoft.Graph.Auth
     /// </summary>
     public class ClientCredentialProvider : IAuthenticationProvider
     {
+        /// <summary>
+        ///  A Scope property
+        /// </summary>
+        public string Scope { get; set; }
```
Comment (darrelmiller) [inline src/Microsoft.Graph.Auth/ConfidentialClient/ClientCredentialProvider.cs:22]: I don't think this should be a single Scope string.  I think it should be an array of scope strings.  The goal is to allow the use of this provider for non-graph APIs.  We can't assume that non-Graph APIs only have a single scope value.

**Comment by pschaeflein** [comment]
I would caution against including .default with the user-provided scope.

If the request includes other resources (say SharePoint Online or custom resource), then including .default for Microsoft Graph is not supported. AAD won't issue a token. (AADSTS70011: .default scope can't be combined with resource-specific scopes) 

I have a real-world use case where the scopes parameter had to be explicit:
```
https://[tenant].sharepoint.com/AllSites.FullControl
https://graph.microsoft.com/Mail.Send
```

**Comment by darrelmiller** [comment]
@pschaeflein I'm not sure I follow.  I don't think we are adding .default to user provided scopes.  If you provide an explicit scope(s), we don't touch it.

**Comment by pschaeflein** [comment]
Then I mis-understood Michael's comment. Never mind. :)

**Review by darrelmiller — APPROVED** [review]

**Review by MIchaelMainer — APPROVED** [review]

**Pull request merged**